storage/src/store/shard_state/mod.rs

```diff
@@ -211,14 +209,26 @@ impl ShardStateStorage {
             }
 
             alloc.reset();
-            let mut batch = rocksdb::WriteBatch::default();
+
             {
                 let _guard = self.gc_lock.lock().await;
-                let total = self
-                    .cell_storage
-                    .remove_cell(&mut batch, &alloc, root_hash)?;
-                batch.delete_cf(&shard_states_cf.bound(), key);
-                raw.write_opt(batch, cells_write_options)?;
+
+                let db = self.db.clone();
+                let cell_storage = self.cell_storage.clone();
+                let key = key.to_vec();
+                let mut batch = rocksdb::WriteBatch::default();
+
+                let total = tokio::task::spawn_blocking(move || {
+                    let total = cell_storage.remove_cell(&mut batch, &root_hash)?;
```

Comment:
Please reuse `alloc` here. You can just move it into and out of the spawn since it is `Send`

Reply:
done